### DISCUSSION TITLE
Birdnet-Go Human Voice Detection?

### DISCUSSION THREAD

**Discussion by London-Lass** [tphakala/birdnet-go#20] [Ideas]
Carrying over a conversation started on Birdnet-PI, where the suggestion was made that it might be possible to use birdnet-go to detect and trigger an action if human voice is heard.  Any thoughts on how this might be implemented?

**Comment by tphakala** [comment]
It requires some new code in real-time detection side, should not be that hard to implement once I have time. I can't promise anything but I will see if I can come up something later this week. I am currently working on web UI to make this app more useful for more people.

**Comment by London-Lass** [comment]
Any further thoughts on this? I think I've worked out how to mute OBS, I just need to be able to raise an HTTPS PUSH/POST to an endpoint to trigger this.

**Reply by tphakala** [reply]
Sorry for lack of progress, I have been working on web UI and it threw me off refactoring lot of old code which has taken all my spare time on this project. I can't still make any promises when this feature will be available but it is still on my todo list.

**Reply by tphakala** [reply]
I have now pushed a code to github which supports species specific custom scripts, by creating ~/.config/birdnet-go/species_config.cvs or /etc/birdnet-go/species_config.cvs and adding

Human vocal,0.3,ExecuteScript:/path/script.sh

would execute script in /path/script.sh any time human vocal is detected, 0.3 in this example is prediction treshold which triggers detection.

**Comment by London-Lass** [comment]
Is there likely to be a download for the latest version at some point for raspberry pi 4? I tried to compile, but getting a bunch of errors.

**Reply by tphakala** [reply]
Yes absolutely, I will upload a binary for RPI 3 & 4 in few days

**Reply by tphakala** [reply]
Binary release with species specific action support is now uploaded to releases

**Reply by London-Lass** [reply]
Is Human voice included in the model by default? I downloaded your binary, it runs just fine, detects birdsong, but doesn't seem to react when I speak? I've created the ~/.config/birdnet-go/species_config.cvs file with the appropriate content.

I've turned on logging, but no reaction to me "testing testing 123"

**Reply by tphakala** [reply]
Yes it is, can you please provide contents of species_config.csv. 

Is it detecting anything if you for example play birdsongs from youtube etc?

**Reply by London-Lass** [reply]
Works perfectly for birds, lots of nice spectograms etc.

`Human vocal,0.3,executeScript:/home/birdnet/.config/birdnet-go/human.sh`

And human.sh is executable and just echos out Yes

**Reply by tphakala** [reply]
Can you drop detection threshold to 0.1?

birdnet-go should report all detections to tty console or journald logs as it runs, so even if you would not have any actions set to human vocals you should still see it detecting it if it exceeds detection threshold

2024/02/17 13:20:46 New detection: Human vocal with confidence: 0.227212

If you do not run birdnet-go in tty console you should see logs in journald with command like
journalctl -n 100 -t birdnet-go

"-n 100" lists last 100 messages
"-t birdnet-go" returns log messages sent by birdnet-go

**Comment by London-Lass** [comment]
Having played around with thresholds, and volume of vocals, i've come to the conclusion that sometimes it hears voices, sometimes it doesn't. I can speak at a level and average volume to it, and it won't hear anything, then I can play it a youtube video with say american accents in it and it hears it. It can hear me whistle no problem. I'm wondering if its due to accents or just the sensitivity of the microphone? Currently using a good quality logitech webcam. When it does register  Human Vocal it's very clear on playback.

**Comment by London-Lass** [comment]
Is there any way to select/change the model file?

**Reply by tphakala** [reply]
At the moment there is no option to change model at runtime as model file(s) are set and embedded into binary at compile time. However you can change model file if you are able to compile birdnet-go yourself. 

Do you have a custom model you would like to use instead of BirdNET v2.4 model?

**Comment by robinsandfort** [comment]
Hi everyone,
did you think about using it in the opposite direction?
Without something like the BirdNET-Pi privacy settings I can´t use 
BirdNET-Go. I would really love to make the move towards BirdNET-Go.
Did anyone use a custom script to remove Human vocal or better never save it 
in the first place. You do a great job with this project Tomi!

Greetings from the woods,
Robin

**Reply by tphakala** [reply]
Hi Robin, BirdNET-Pi like privacy feature is in plans, should be quite easy to implement so I will likely do it during next weekend or so.

**Reply by robinsandfort** [reply]
That sounds wonderful Tomi!
Greetings,
Robin

**Reply by tphakala** [reply]
Privacy filter is now implemented and included github codebase.

If you are able to compile code yourself you can enable privacy filter by adding privacyfilter enabled setting under realtime setting like

```
realtime:
  privacyfilter:
    enabled: true
```
